### PR TITLE
PDE-5636 feat(cli): include extra fields for analytics 

### PR DESCRIPTION
**Pull request by Natay**
This pull request expands the analytics functionality in the Zapier CLI by updating the way arguments are recorded and adding additional information to the analytics payload.

#### Changes
- **Updated `analytics.js`**:
  - Included `app_id` in the analytics payload, which is fetched from the linked app configuration OR integrationid" in arguments since some commands ( like "zapier convert" ) won't have a linked app directory when called. 
  
- **Updated `ZapierBaseCommand.js`**:
    - Change `Object.keys(this.args)` to `this.args`. This allows us to later capture app ID argument value based on the name and populate `numArgs` and parts of `flags` . This is not meant to capture `this.args` in analytics as shown in tests below 

#### Impact
The inclusion of app ID will provide better insights into CLI usage patterns. There is another ticket to track work required for adding "arguments" effectively and safely PDE-5641

#### Tests
Tested the changes locally

Screenshot 2024-12-23 at 9 53 27 AM

```
➜  zapier-platform git:(PDE-5636_expand_analyticsBody) ✗ zapier-dev convert 216691 testing5 -v 1.0.0 --debug  
(node:72410) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
⠋ Downloading integration from Zapier  zapier:analytics sending {
  command: 'convert',
  isValidCommand: true,
  numArgs: 2,
  app_id: 216691,
  flags: { version: '1.0.0', debug: true },
  cliVersion: '16.0.0',
  os: 'darwin',
  sendUserId: true
} +0ms
⠙ Downloading integration from Zapier  zapier:api >> POST undefined +0ms
  zapier:api >> {"command":"convert","isValidCommand":true,"numArgs":2,"app_id":216691,"flags":{"version":"1.0.0","debug":true},"cliVersion":"16.0.0","os":"darwin"} +1ms
  zapier:api << 200 +0ms
  zapier:api <<  +0ms
  zapier:api ------------ +0ms
  zapier:analytics success: true +122ms
  zapier:api >> GET undefined +16ms
  zapier:api << 200 +0ms
  zapier:api <<  +0ms
  zapier:api ------------ +0ms
⠹ Downloading integration from Zapier  zapier:api >> GET undefined +56ms
  zapier:api << 200 +0ms
  zapier:api <<  +0ms
  zapier:api ------------ +0ms
✔ Downloading integration from Zapier
✔ Writing .env
✔ Writing package.json
✔ Writing .zapierapprc
✔ Writing index.js
✔ Writing authentication.js
✔ Writing triggers/test_key.js
✔ Writing test/triggers/test_key.test.js
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/.env
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/.gitignore
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/.zapierapprc
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/authentication.js
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/index.js
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/package.json
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/triggers/test_key.js
✔ Copying /Users/natay/src/github.com/zapier-platform/testing5/test/triggers/test_key.test.js
```

### CODE REVIEW

File: packages/cli/src/utils/analytics.js [hunk 1]
```diff
@@ -20,24 +21,26 @@ const shouldSkipAnalytics = (mode) =>
...
-
+  // We don't want to "explode" if appID is missing
+  const linkedAppId = (await getLinkedAppConfig(undefined, false))?.id;
   // to make this more testable, we should split this out into its own function
   const analyticsBody = {
     command,
     isValidCommand,
-    numArgs: argNames.length,
+    numArgs: argValues.length,
+    arguments: argValues,
```
Comment (Natay) [inline packages/cli/src/utils/analytics.js:39]: Open question: should we remove some arguments ( like path ) from this?

Comment (Natay) [reply packages/cli/src/utils/analytics.js]: Removing `arguments` altogether ✅

File: packages/cli/src/utils/analytics.js [hunk 2]
```diff
@@ -20,24 +21,38 @@ const shouldSkipAnalytics = (mode) =>
...
 
+  const integrationIDKey = argKeys.find(
+    (key) => key.toLowerCase() === 'integrationid',
+  );
+  const integrationID = integrationIDKey ? args[integrationIDKey] : undefined;
+
+  // Some commands ( like "zapier convert" ) won't have an app directory when called.
+  // Instead, having the app ID in the arguments.
+  // In this case, we fallback to using "integrationid" in arguments ( if it's there )
+  // and don't want to "explode" if appID is missing
```
Comment (eliangcs) [inline packages/cli/src/utils/analytics.js:43]: Nice catch!

**Review by eliangcs — DISMISSED** [review]
I did some quick tests to make sure commands still work. And they do! Approving.

However, while testing, I noticed the debug logs print `undefined` request URLs.



This may be not related to this PR. But maybe you can fix it along the way? You can fix it by replacing this line:

https://github.com/zapier/zapier-platform/blob/a0109d001921762f725918c1aeb01d3712e9b3a7/packages/cli/src/utils/api.js#L98

with:

```
  debug(`>> ${requestOptions.method} ${requestOptions.url || res.url}`);
```

**Comment by Natay** [comment]
Thanks for the review and help testing @eliangcs! Followed your suggestion and fixed the debug log!

**Review by eliangcs — APPROVED** [review]
Thanks for the change! Approved.

**Review by rnegron — APPROVED** [review]

**Pull request merged**